tsconfig.base.json

```diff
@@ -0,0 +1,41 @@
+{
+  "compilerOptions": {
+    "module": "esnext",
+    "target": "ES2015",
+    "lib": ["es2017", "dom"],
```

Comment:
Unless we plan to start requiring consumers to use polyfills, we should only include up to es5 to ensure IE 11 support. (We most likely can't build our code itself with polyfills included due to bundle size concerns.)

Reply:
`lib` specification from TypeScript is not accurate btw - for that reason, babel should be used to properly transpile code based on browsers we want support, thus not blocking us to not be able to use latest ECMAscript spec.

> Unless we plan to start requiring consumers to use polyfills, we should only include up to es5 to ensure IE 11 support. 

Do we have somewhere proper set of requirements what is needed/ what we plan to support for convergence  ?

Reply:
> lib specification from TypeScript is not accurate btw - for that reason, babel should be used to properly transpile code based on browsers we want support, thus not blocking us to not be able to use latest ECMAscript spec.

What do you mean by not accurate? I know IE 11 supports a few things beyond what `lib: es5` includes, but for v7/8 it's generally been an adequate heuristic.

> > Unless we plan to start requiring consumers to use polyfills, we should only include up to es5 to ensure IE 11 support.
>
> Do we have somewhere proper set of requirements what is needed/ what we plan to support for convergence ?

This has been the perpetual headache of the past year: people from one team have one set of requirements in their minds, and the other team has a slightly different set, and often nobody notices until pretty late (when we've already wasted months of work) because nobody wrote the requirements down.

In this case, I'm not 100% clear on what the requirements are since I've only been on the edge of convergence discussions. I posted about it in the convergence channel and will try to get someone who's more closely involved to write it down.

Reply:
> What do you mean by not accurate?

babel can apply transpilation set by proper browser support restrictions. In case of ie 11, `es5` is too strict and low level -> thus increasing bundle size/perf etc. If we setup browser restrictions accordingly we can  produce more concise code to be shipped to consumers.